src/utils/test-utils.tsx

```diff
@@ -16,31 +22,42 @@ const goTo = (route?: string, name?: string) => {
     );
 };
 
-const waitForLoading = async () => {
-    return Promise.resolve();
-};
+// TODO - We use immer here but don't need it. The user metadata is currently just
+//  any number of keys (according to the types). Hoping that changes in the future.
+const generateMockUserMetadata = (username: string) => {
+    return produce(mockDeep<User['user_metadata']>(), (draft) => {
+        draft.avatar_url = `https://example.org/avatar/${username}`;
+        draft.email = `${username}@example.org`;
+        draft.email_verified = true;
+        draft.full_name = `Full ${username}`;
+        draft.iss = 'https://api.example.org';
+        draft.name = username;
+        draft.picture = `https://example.org/picture/${username}`;
+        draft.preferred_username = username;
+        draft.provider_id = '00000000000000000000_provider';
+        draft.sub = '00000000000000000000_sub';
+        draft.user_name = username;
```

Comment:
Right now we just use the username in as many things as it makes sense to so we can check for them in the UI. Eventually we can add ability to manually set more variables in the future if it is needed.